### PR TITLE
Iss14 dmx channel on and off

### PR DESCRIPTION
**Pull request by bookdude13**
DMX page now can set individual channels, ranges, boxes, and all to be on or off. 

@theeldestelder @ByronAmbright How were you guys thinking about doing the fades? We could try to do a slider button, or just continuous fading on a range/channel like you did in the demo?

Note: the layout part of #14 was not done, since that isn't implemented yet afaik.

### CODE REVIEW

**Comment by bookdude13** [comment]
Alrighty, fades added in. Ready for review

**Comment by theeldestelder** [comment]
I was thinking just a continuous fade across all channels with an on/off button. Were you thinking a slider to change the fade rate faster/slower?  That might be valuable too.  I know last year we wanted the fades to test the box flickering issue so maybe being able to vary the fades make sense?

I'll get to looking at this tonight and have a review for sure by the morning.

**Comment by bookdude13** [comment]
I talked to Byron and got clarification. I wasn't sure if the fades mentioned in the issue referred to some sort of slider to control a range of channels from 0-255. This PR is currently two buttons doing a continuous fade. I like the idea of an adjustable fade, but that might be worth making as a separate issue/feature to add.

File: app/js/dmx.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+// Create a universe
+const universe = dmx.addUniverse('demo', DRIVER, SERIAL_PORT);
+
+// Sets a range of channels to be on or off. Range is inclusive on both ends
+// channelStart is an int, channelEnd is an int, value is an int in range [0, 255]
+// Channels are 0-indexed
+function setRange(channelStart, channelEnd, value) {
+    const channels = {};
+    for (let i = channelStart; i <= channelEnd; i += 1) {
+        // DMX is 1-indexed. This should be the only place where the +1 is added
```
Comment (theeldestelder) [inline app/js/dmx.js]: I sure hope so lol

File: app/html/dmx.html [hunk 2]
```diff
@@ -2,11 +2,158 @@
...
+            setRange(start, end, 255);
+        };
+        document.getElementById('rangeOff').onclick = function () {
+            const start = getIntFromElementById("rangeStart", -1);
+            if (start == -1) return;
+
+            const end = getIntFromElementById("rangeEnd", -1);
+            if (end == -1) return;
+
+            setRange(start, end, 0);
```
Comment (theeldestelder) [inline app/html/dmx.html]: Should this prompt the user if start > end or either of them are negative numbers?

**Comment by theeldestelder** [comment]
Interesting that the Windows job failed.  That should get looked at once we have some spare time.

File: app/html/dmx.html (hunk 2 above)
Comment (bookdude13) [reply app/html/dmx.html]: Probably. Will add that check

**Comment by bookdude13** [comment]
I can do a couple style/UI changes quick for this. Also not sure about the Windows build. I think I accidentally upgraded electron and a couple other libraries in the process of installing electron-rebuild, so maybe that's causing it?

**Comment by bookdude13** [comment]
It looks like electron-rebuild (which I pulled in to handle serialport not being built to electron's node version) depends on python2. Adding that to the azure pipeline fixed the issue. Ready for re-review, since there's a lot of changes I made (organization is better now).

**Comment by bookdude13** [comment]
I have some stuff ready for iss13 once this gets approved (it's based on these changes, since it touches dmx_output.js and uses serialport, which requires electron-rebuild)

**Comment by bookdude13** [comment]
Closes #14. Will make new issue for layout part

**Pull request merged**